crates/sui-types/src/base_types.rs

```diff
@@ -218,6 +219,7 @@ impl<T: SuiPublicKey> From<&T> for SuiAddress {
         let g_arr = hasher.finalize();
 
         let mut res = [0u8; SUI_ADDRESS_LENGTH];
+        // TODO: should this panic?
```

Comment:
Ditto, // OK to access slice because Sha3_256 should never be shorter than SUI_ADDRESS_LENGTH.